math/vc_soa/include/algebra/math/impl/vc_soa_getter.hpp

```diff
@@ -0,0 +1,122 @@
+/** Algebra plugins library, part of the ACTS project
+ *
+ * (c) 2022-2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "algebra/qualifiers.hpp"
+#include "algebra/storage/vector.hpp"
+
+// Vc include(s).
+#ifdef _MSC_VER
+#pragma warning(push, 0)
+#endif  // MSVC
+#include <Vc/Vc>
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif  // MSVC
+
+// System include(s)
+#include <cmath>
+#include <type_traits>
+
+namespace algebra::vc_soa::math {
+
+/// This method retrieves phi from a vector, vector base with rows >= 2
+///
+/// @tparam N dimension of the vector
+/// @tparam scalar_t scalar type
+/// @tparam array_t array type that holds the vector elements
+///
+/// @param v the input vector
+template <std::size_t N, typename scalar_t,
+          template <typename, std::size_t> class array_t,
```

Comment:
In the past it hurt us a couple of times when we hardcoded the "size variable" of an array type to be `std::size_t`. (Since some libraries use `unsigned int` instead.) Is that not expected to become a problem here? :thinking:

Reply:
And even with ```Vc::SimdArray``` it needs an alias type which seems to confuse the MSVC compiler maybe? I am open to suggestions here